administrator/components/com_languages/tmpl/override/edit.php

```diff
@@ -22,23 +25,8 @@
 JHtml::_('behavior.core');
 JHtml::_('jquery.framework');
 JHtml::_('script', 'com_languages/overrider.min.js', array('version' => 'auto', 'relative' => true));
-
-JFactory::getDocument()->addScriptDeclaration('
-	jQuery(document).ready(function($) {
-		$("#jform_searchstring").on("focus", function() {
-			if (!Joomla.overrider.states.refreshed)
-			{
-				var expired = "' . $expired . '";
-				if (expired)
-				{
-					Joomla.overrider.refreshCache();
-					Joomla.overrider.states.refreshed = true;
-				}
-			}
-			$(this).removeClass("invalid");
-		});
-	});
-');
+HTMLHelper::_('script', 'com_languages/admin-override-refresh-searchstring.js', ['relative' => true, 'version' => 'auto']);
+Factory::getDocument()->addScriptOptions('search-string-expired', $expired);
```

Comment:
i prefer to use data-attributes to pass data from php to js. We should not use scriptOptions for everything, because sooner or later we will get collusions or conflicts. i.e. 2 scripts use the same key in script options.  

You can add the data attribute to #jform_searchstring

Reply:
The JFormField (#jform_searchstring) of Type Text does not support the data-* attribute, right? How can I add it? Shall I create a custom JFormField for supporting the attribut?

Reply:
There is a pr in staging for data-*